app/code/Magento/Catalog/etc/adminhtml/system.xml

```diff
@@ -158,6 +161,7 @@
                 </field>
                 <field id="year_range" translate="label comment" type="text" sortOrder="4" showInDefault="1" showInWebsite="1" showInStore="1">
                     <label>Year Range</label>
+                    <validate>validate-digits validate-zero-or-greater validate-number-range number-range-1000-9999</validate>
```

Comment:
I am not sure if it help.
I am still able to save wrong value.
![image](https://user-images.githubusercontent.com/9420807/62142511-26417100-b2f7-11e9-9635-6637fe82c974.png)
Could you please check if this can be fixed?

Reply:
![validateok](https://user-images.githubusercontent.com/8234209/62143895-0d4ab700-b31b-11e9-80d5-1e496ea247eb.gif)

I didn't see the problem. Could you please check it again? Thanks

Reply:
@swnsma

Reply:
Yes, seem it works.